Implemented `gsw` so it uses `git switch`

LittleHaku opened the pull request:
<!-- NOTE: forgit.plugin.zsh & forgit.plugin.sh share the same code. You should make sure the changes work in both `zsh` & `bash` -->

<!-- Check all that apply [x] -->

As discussed #447, we should use the new `git switch` and `git restore` instead of using `git checkout` for everything.

In this PR I implemented the `gsw` with `git switch` and `gcb` now calls this new function and echoes a deprecated warning.

I haven't done the git restore yet since I want to be sure that this is properly done before going for a new one.

## Check list

- [x] I have performed a self-review of my code
- [x] I have commented my code in hard-to-understand areas
- [ ] I have added unit tests for my code
- [x] I have made corresponding changes to the documentation

## Description

<!-- Please include a summary of the change (and the related issue, if any). Please also include relevant motivation and context when necessary. -->

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Refactor
- [x] Breaking change
- [ ] Test
- [ ] Documentation change

## Test environment

- Shell
    - [x] bash
    - [x] zsh
    - [ ] fish
- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Others:

sandr01d commented on bin/git-forgit:
```diff
@@ -849,14 +849,18 @@ _forgit_checkout_file() {
-# git checkout-branch selector
 _forgit_checkout_branch() {
+    _forgit_switch_branch "$@"
+    echo "forgit: 'gcb' is deprecated and will be removed in a future release. Please use 'gsw' instead." >&2
```
What I meant with my comment regarding the removal of `gcb` is that I do not want to entire `_forgit_checkout_branch` function to stick around forever, which would result in two separate implementations for the same thing. Keeping the old alias is fine with me.

sandr01d reviewed (changes requested):
Thanks for the PR @LittleHaku! Looks mostly good, just a few things to address (see my other comments). I noticed that the completions for fish are currently missing.

LittleHaku commented:
> Thanks for the PR @LittleHaku! Looks mostly good, just a few things to address (see my other comments). I noticed that the completions for fish are currently missing.

I already added them :)

LittleHaku commented:
> > > Thanks for the PR @LittleHaku! Looks mostly good, just a few things to address (see my other comments). I noticed that the completions for fish are currently missing.
> > 
> > 
> > I already added them :)
> 
> Thanks for keeping things going @LittleHaku. Since nobody complained so far, I'd say we bring back the old `gcb` for now and merge this PR soon. Removing it later is easy.

Just added the original `gcb` back @sandr01d
